bluemira/power_cycle/timeline.py

```diff
@@ -0,0 +1,306 @@
+"""
+Classes to define the timeline for Power Cycle simulations.
+"""
+
+# Import
+from typing import Union  # , List
+
+from bluemira.power_cycle.base import PowerCycleABC as imported_abc
+from bluemira.power_cycle.base import PowerCycleError as imported_error
+from bluemira.power_cycle.base import PowerCycleUtilities as imported_utilities
+from bluemira.power_cycle.base import classproperty
+
+# ######################################################################
+# POWER CYCLE PHASE
+# ######################################################################
+
+
+class PowerCyclePhase(imported_abc):
+    """
+    Class to define phases for a Power Cycle pulse.
+
+    Parameters
+    ----------
+    name: 'str'
+        Description of the `PowerCyclePhase` instance.
+    label: `str`
+        Shorthand label for addressing the `PowerCyclePhase` instance.
+    dependency: `str`
+        Classification of the `PowerCyclePhase` instance in terms of
+        time-dependent calculation: 'ss' (stready-state) or 'tt'
+        (transient).
+    duration: `float`
+        Phase duration. [s]
+    """
+
+    # ------------------------------------------------------------------
+    # CLASS ATTRIBUTES
+    # ------------------------------------------------------------------
+
+    _valid_dependencies = {
+        "ss": "steady-state",
+        "tt": "transient",
+    }
+
+    # Error messages
+    @classproperty
+    def _errors(cls):
```

Comment:
I doubt this is needed, call the errors where they're needed, later maybe we can do something like this if it is needed.